<a href="https://colab.research.google.com/github/Maleewij65/heart-disease-prediction-ML_Assignment/blob/IT22339324_Catboost/IT22339324_Catboost.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# CatBoost Classification Model

This component applies the CatBoost machine learning algorithm for heart disease
prediction using the Heart Disease dataset. Medical attributes such as age,
cholesterol level, blood pressure, and heart rate are used as input features,
while the target variable represents the presence or absence of heart disease.

CatBoost is a supervised machine learning algorithm based on gradient boosting
decision trees. It is selected because it performs well on structured tabular
data, reduces overfitting, and typically achieves higher accuracy compared to
traditional classifiers such as Logistic Regression.

In [2]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
!pip install catboost

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 97.1/97.1 MB 8.6 MB/s eta 0:00:00


In [4]:
# Core libraries
import pandas as pd
import numpy as np

# Visualization
import matplotlib.pyplot as plt
import seaborn as sns

In [5]:
data = pd.read_csv('/content/drive/MyDrive/heart.csv')

print("Dataset loaded successfully")
print(data.shape)
data.head()

Dataset loaded successfully
(918, 12)


,Age,Sex,ChestPainType,RestingBP,Cholesterol,FastingBS,RestingECG,MaxHR,ExerciseAngina,Oldpeak,ST_Slope,HeartDisease
0,40,M,ATA,140,289,0,Normal,172,N,0.0,Up,0
1,49,F,NAP,160,180,0,Normal,156,N,1.0,Flat,1
2,37,M,ATA,130,283,0,ST,98,N,0.0,Up,0
3,48,F,ASY,138,214,0,Normal,108,Y,1.5,Flat,1
4,54,M,NAP,150,195,0,Normal,122,N,0.0,Up,0


### Dataset Description

The dataset used in this project is the **Heart Disease Dataset**, which contains medical records of patients used to predict the presence of heart disease. The dataset includes several clinical and physiological attributes such as age, resting blood pressure, cholesterol level, maximum heart rate achieved, and chest pain type.

The dataset consists of **918 patient records** and **12 attributes**, where **11 features** are used as input variables and **1 target variable (HeartDisease)** indicates whether the patient has heart disease or not.

The target variable **HeartDisease** is binary:
- **0** – No heart disease
- **1** – Presence of heart disease

The dataset contains a mix of **numerical features** (such as Age, Cholesterol, RestingBP, MaxHR, and Oldpeak) and **categorical features** (such as Sex, ChestPainType, RestingECG, ExerciseAngina, and ST_Slope). This combination of feature types makes the dataset suitable for applying machine learning classification algorithms to predict heart disease risk.

Data Exploration

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 918 entries, 0 to 917
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Age             918 non-null    int64  
 1   Sex             918 non-null    object 
 2   ChestPainType   918 non-null    object 
 3   RestingBP       918 non-null    int64  
 4   Cholesterol     918 non-null    int64  
 5   FastingBS       918 non-null    int64  
 6   RestingECG      918 non-null    object 
 7   MaxHR           918 non-null    int64  
 8   ExerciseAngina  918 non-null    object 
 9   Oldpeak         918 non-null    float64
 10  ST_Slope        918 non-null    object 
 11  HeartDisease    918 non-null    int64  
dtypes: float64(1), int64(6), object(5)
memory usage: 86.2+ KB


In [7]:
data.describe()

,Age,RestingBP,Cholesterol,FastingBS,MaxHR,Oldpeak,HeartDisease
count,918.000000,918.000000,918.000000,918.000000,918.000000,918.000000,918.000000
mean,53.510893,132.396514,198.799564,0.233115,136.809368,0.887364,0.553377
std,9.432617,18.514154,109.384145,0.423046,25.460334,1.066570,0.497414
min,28.000000,0.000000,0.000000,0.000000,60.000000,-2.600000,0.000000
25%,47.000000,120.000000,173.250000,0.000000,120.000000,0.000000,0.000000
50%,54.000000,130.000000,223.000000,0.000000,138.000000,0.600000,1.000000
75%,60.000000,140.000000,267.000000,0.000000,156.000000,1.500000,1.000000
max,77.000000,200.000000,603.000000,1.000000,202.000000,6.200000,1.000000


In [11]:
print("Missing values in each column:")
data.isnull().sum()

Missing values in each column:


,0
Age,0
Sex,0
ChestPainType,0
RestingBP,0
Cholesterol,0
FastingBS,0
RestingECG,0
MaxHR,0
ExerciseAngina,0
Oldpeak,0
